File: Liskovs/violations/ShapeFactory.java
class ShapeFactory {
    public Shape createShape() {
        // The post condition is that it always returns a shape
        return new Shape();
    }
    public void draw () {
        System.out.println("Drawing");
    }
}

class UnsafeShapeFactory extends ShapeFactory {
    @Override
    public Shape createShape() {
        // Violates parent class contract - It returns a non shape object
        return null;
    }
}

File: Liskovs/violations/ShapeFactory.cs
using System;

public class Shape
{
    // Empty shape class for demonstration
}

public class ShapeFactory
{
    public virtual Shape CreateShape()
    {
        // Post-condition: always returns a Shape
        return new Shape();
    }

    public void Draw()
    {
        Console.WriteLine("Drawing");
    }
}

public class UnsafeShapeFactory : ShapeFactory
{
    public override Shape CreateShape()
    {
        // Violates parent class contract and returns null
        return null;
    }
}
